### PR TITLE
Implemented Polygon.collidepoint()

### PR DESCRIPTION
**Pull request by itzpr3d4t0r**
One of the key features of shapes, allows you to do:

```Python
poly.collidepoint((x, y))
# or
poly.collidepoint(x, y)
```

### CODE REVIEW

File: src_c/polygon.c [hunk 1]
```diff
@@ -460,9 +460,38 @@ pg_polygon_move_ip(pgPolygonObject *self, PyObject *const *args,
...
+        }
+    }
+    else if (nargs == 2) {
+        if (!pg_DoubleFromObj(args[0], &x) || !pg_DoubleFromObj(args[1], &y)) {
+            goto error;
+        }
+    }
+    else {
+        goto error;
+    }
```
Comment (Emc2356) [inline src_c/polygon.c:481]: so far we have used this code a lot of times, i think it is time to put it as a function in `geometry.c`/`geometry.h` and name it like _pgTwoDoublesFromFastcall

File: src_c/collisions.c [hunk 2]
```diff
@@ -289,3 +289,29 @@ pgCollision_RectCircle(SDL_Rect *rect, pgCircleBase *circle)
...
+        double yj = poly->vertices[j * 2 + 1];
+
+        if (((yi > y) != (yj > y)) &&
+            (x < (xj - xi) * (y - yi) / (yj - yi) + xi)) {
+            collision = !collision;
+        }
+    }
+
+    return collision;
+}
```
Comment (Emc2356) [inline src_c/collisions.c:317]: you need a newline here

**Review by Emc2356 — APPROVED** [review]
overall looks great

File: src_c/polygon.c (hunk 1 above)
Comment (itzpr3d4t0r) [reply src_c/polygon.c:481]: I think we should do that just not in this PR.

Comment (itzpr3d4t0r) [reply src_c/polygon.c:481]: I'll open another PR that adds that and applies that where necessary.

**Pull request merged**